pkg/apis/core/validation/utils.go

```diff
@@ -52,15 +52,25 @@ func validateSecretReference(ref corev1.SecretReference, fldPath *field.Path) fi
 func validateCrossVersionObjectReference(ref autoscalingv1.CrossVersionObjectReference, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	if len(ref.APIVersion) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("apiVersion"), "must provide a apiVersion"))
+	} else {
+		if ref.APIVersion != corev1.SchemeGroupVersion.String() {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("apiVersion"), ref.APIVersion, []string{corev1.SchemeGroupVersion.String()}))
+		}
+	}
+
 	if len(ref.Kind) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("kind"), "must provide a kind"))
+	} else {
+		if ref.Kind != "Secret" && ref.Kind != "ConfigMap" {
```

Comment:
Unit tests missing

Reply:
`validateCrossVersionObjectReference` is not exported, so can't add unit test.

Reply:
You can add a test in `core/validation/shoot.go`